PostalServerDotNet/v1/Client.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using PostalServerDotNet.v1.Model.Response;
+using RestSharp;
+using System;
+using System.Net;
+
+namespace PostalServerDotNet.v1
+{
+    public partial class Client
+    {
+        #region Properties
+
+        /// <summary>
+        /// The client
+        /// </summary>
+        private readonly IRestClient _client;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Client"/> class.
+        /// </summary>
+        /// <param name="baseUrl">The baseUrl for the API service.</param>
+        /// <param name="apiKey">The API key.</param>
+        public Client( string baseUrl, string apiKey )
+        {
+            _client = new RestClient( baseUrl );
+            _client.AddDefaultHeader( "X-Server-API-Key", apiKey );
+        }
+
+        #endregion
+
+        #region Method
+
+        /// <summary>
+        /// Executes the specified request.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="request">The request.</param>
+        /// <returns></returns>
+        public T Execute<T>( RestRequest request ) where T : new()
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+            var response = _client.Execute<T>( request );
+
+            if ( response.Content.Contains( "error" ) )
```

Comment:
I think we need to make sure to handle actual errors as well, just in case there is an http error and the api doesn't actually respond with a proper error.

```suggestion
            if ( response.ErrorException != null )
            {
                const string message = "Error retrieving response.  Check inner details for more info.";
                var exception = new ApplicationException( message, response.ErrorException );
                throw exception;
            }
            else if ( response.Content.Contains( "error" ) )
```